test/test_encoders.py

```diff
@@ -1084,9 +1130,51 @@ def test_codec_spec_vs_impl_equivalence(self, tmp_path, codec_spec, codec_impl):
             dest=impl_output, codec=codec_impl
         )
 
-        assert self._get_codec_spec(spec_output) == codec_spec
-        assert self._get_codec_spec(impl_output) == codec_spec
+        assert (
+            self._get_video_metadata(spec_output, fields=["codec_name"]).get(
+                "codec_name"
+            )
+            == codec_spec
+        )
+        assert (
+            self._get_video_metadata(impl_output, fields=["codec_name"]).get(
+                "codec_name"
+            )
+            == codec_spec
+        )
 
         frames_spec = self.decode(spec_output).data
         frames_impl = self.decode(impl_output).data
         torch.testing.assert_close(frames_spec, frames_impl, rtol=0, atol=0)
+
+    @pytest.mark.skipif(in_fbcode(), reason="ffprobe not available")
+    @pytest.mark.parametrize(
+        "profile,colorspace,color_range",
+        [
+            ("baseline", "bt709", "tv"),
+            ("main", "bt470bg", "pc"),
+            ("high", "fcc", "pc"),
+        ],
+    )
+    def test_extra_options_utilized(self, tmp_path, profile, colorspace, color_range):
+        # Test setting profile, colorspace, and color_range via extra_options is utilized
+        source_frames = torch.zeros((5, 3, 64, 64), dtype=torch.uint8)
+        encoder = VideoEncoder(frames=source_frames, frame_rate=30)
+
+        output_path = str(tmp_path / "output.mp4")
+        encoder.to_file(
+            dest=output_path,
+            extra_options={
+                "profile": profile,
+                "colorspace": colorspace,
+                "color_range": color_range,
+            },
+        )
+        metadata = self._get_video_metadata(
+            output_path,
+            fields=["profile", "color_space", "color_range"],
+        )
+        # Validate profile (case-insensitive, baseline is reported as "Constrained Baseline")
+        assert profile in metadata.get("profile", "").lower()
```

Comment:
I'd suggest the following, because using `in` is less strict and can sometimes be surprisingly easy to pass, like `assert "" in "abc"`


```suggestion
        expected_profile = "constrained baseline" if profile == "baseline" else profile
        assert metadata["profile"].lower() == expected_profile
```